src/main/java/uk/nhs/digital/nhsconnect/lab/results/translator/mapper/PractitionerMapper.java

```diff
@@ -13,20 +14,45 @@
     private static final String SDS_USER_SYSTEM = "https://fhir.nhs.uk/Id/sds-user-id";
 
     public Optional<Practitioner> mapRequester(final Message message) {
-        return message.getInvolvedParties().stream()
+        Optional<RequesterNameAndAddress> requester = message.getInvolvedParties().stream()
             .map(InvolvedParty::getRequesterNameAndAddress)
             .flatMap(Optional::stream)
-            .map(this::toPractitioner)
             .findAny();
+
+        if (requester.isPresent()) {
+            final var r = requester.get();
+            Practitioner practitioner = mapToPractitioner(r.getIdentifier(), r.getRequesterName());
+            return Optional.of(practitioner);
+        } else {
+            return Optional.empty();
+        }
     }
 
-    private Practitioner toPractitioner(final RequesterNameAndAddress requester) {
+
+    public Optional<Practitioner> mapPerformer(final Message message) {
+        Optional<PerformerNameAndAddress> performer = message.getInvolvedParties().stream()
+            .map(InvolvedParty::getPerformerNameAndAddress)
+            .flatMap(Optional::stream)
+            .findAny();
```

Comment:
`InvolvedParty::getPerformerNameAndAddress` will retrieve all the `NAD+SLA` segments, which includes the performer and the performing organization.
So how can we be sure here that we'll always retrieve a performer here (and not a performing organization)?

I think we'll need to add a `filter`, something like `.filter(performer -> !performer.getIdentifier().isBlank())` (as a performing organization won't have an `identifier`)

Reply:
Done in 5b7f289